src/Storefront/Resources/views/storefront/component/line-item/element/image.html.twig

```diff
@@ -35,14 +35,23 @@
         {% endif %}
     {% endset %}
 
-    {% if productLink %}
-        <a href="{{ seoUrl('frontend.detail.page', { 'productId': lineItem.referencedId }) }}"
+    {# @deprecated productLink variable will be removed without replacement, directly set lineItemLink and lineItemModalLink #}
```

Comment:
I agree that it should be easier to change this. 👍 

I guess just adding a block around `<a>` would not be 100% suitable because it would overwrite it globally? I assume when building an own line-item it is better to pass the desired links when including the label or image.

What do you think of the idea of keeping the variable `productLink` and setting the default `seoUrl` when it is passed as explicit `true`? (Like it is done right now) This could be the default behaviour, and optionally it is possible top pass a custom `productLink` as string like with `lineItemLink`. Then we would not introduce another variable with deprecation. Just the new variable for modal links.

```twig

{# Use default product url when productLink is set to true #}
{% if productLink === true %}
    {% set lineItemLink = seoUrl('frontend.detail.page', { 'productId': lineItem.referencedId }) %}
{% endif %}

{# If ...
- we are on the confirm page
- and do not have an overwrite of the modal link
-> use the default quickview
#}

{% if controllerAction is same as('confirmPage') and productModalLink is not defined %}
    {% set productModalLink = path('widgets.quickview.minimal', { 'productId': lineItem.referencedId }) %}
{% endif %}
```

Reply:
> I guess just adding a block around <a> would not be 100% suitable because it would overwrite it globally?

Yes I think this is not flexible enough, as different plugins might want to change the general appearance, and I as a plugin developer only want to pass the information in a general way, such that a theme designer has the freedom to style this line item as he desires. Currently this would lead to a lot of inconsistencies.

> What do you think of the idea of keeping the variable productLink and setting the default seoUrl when it is passed as explicit true?

At first I wanted to implement it like this as well. But I thought it might be better (in the long run) to have components which do one thing, and try to keep the dependency on the explicit line item types to a minimum. Hence I wanted to deprecate the `productLink` variable.  

> possible top pass a custom `productLink` as string like with `lineItemLink`

This was one approach I also implemented at first, but decided against it, as the naming would also be misleading.

> Then we would not introduce another variable with deprecation. Just the new variable for modal links.

I understand the desire to avoid deprecations, but for the long run it might be more reasonable to implement it in a more general way?

```
{% if controllerAction is same as('confirmPage') and productModalLink is not defined %}
    {% set productModalLink = path('widgets.quickview.minimal', { 'productId': lineItem.referencedId }) %}
{% endif %}
```

It might be, that this piece of code introduces more problems, think for example when I do not want to provide a modal link (e.g. the endpoint does not support ajax modal rendering) on the confirm page, but a link to the item itself. This would lead to a problem, as now an invalid (additional) modal link is generated.

Reply:
> Yes I think this is not flexible enough, as different plugins might want to change the general appearance, and I as a plugin developer only want to pass the information in a general way, such that a theme designer has the freedom to style this line item as he desires. Currently this would lead to a lot of inconsistencies.

👍 

> At first I wanted to implement it like this as well. But I thought it might be better (in the long run) to have components which do one thing, and try to keep the dependency on the explicit line item types to a minimum. Hence I wanted to deprecate the productLink variable.

I see your point, probably it could also be a non-product line item like discount or some special other item which wants to link somewhere. In this case lineItem naming would be more general. It was named productLink because it is only used for the product at the moment. I think it is reasonable to name it differently and deprecate productLink for 6.6.

> It might be, that this piece of code introduces more problems, think for example when I do not want to provide a modal link (e.g. the endpoint does not support ajax modal rendering) on the confirm page, but a link to the item itself. This would lead to a problem, as now an invalid (additional) modal link is generated.

I think the condition `not lineItemLink` was missing in my example 😆

But isn't this a bit weird anyway at the moment? Because the ajax-modal-url is within the the `<a>`, which is only there when the normal link is also provided. But as soon as the modal link is provided, the normal link has no effect anymore. Except the JS is broken or something. Do you think an else-if would be better to explicitly split those cases? So only one of them should be used. I guess this could still be abused by passing both - but this would make the intention maybe a bit more clear that it is either one or the other. 

```twig
{% if lineItemLink %}
    <a href="{{ seoUrl('frontend.detail.page', { 'productId': lineItem.referencedId }) }}">
        {{ label|u.truncate(60, '...', false)|raw }}
    </a>
{% elseif lineItemModalLink %}
    <a href="{{ lineItemModalLink }}"
       data-ajax-modal="true"
       data-url="{{ lineItemModalLink }) }}">
     {{ label|u.truncate(60, '...', false)|raw }}
    </a>
{% else %}
    <div class="line-item-label">
        {{ label|u.truncate(60, '...', false)|raw }}
    </div>
{% endif %}
```

Reply:
And what should the implementation look like when `productLink` is finally removed? Because only checking for `not lineItemLink` will always add the links. When we remove this logic here and directly pass lineItemLink or lineItemModalLink it would make sense to explain inside the deprecation notice that this code block should be removed in favor of something else.

Reply:
Thank you for the suggestion, I implemented it.